sdk/src/assertions/soft_binding.rs

```diff
@@ -0,0 +1,108 @@
+    /// taken from the [C2PA soft binding algorithm list](https://github.com/c2pa-org/softbinding-algorithm-list).
+    ///
+    /// If this field is absent, the algorithm is taken from the `alg_soft` value of the enclosing structure.
+    /// If both are present, the field in this structure is used. If no value is present in any of these places,
+    /// this structure is invalid; there is no default.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub alg: Option<String>,
+
+    /// A list of details about the soft binding.
+    pub blocks: Vec<SoftBindingBlockMap>,
+
+    /// Zero-filled bytes used for filling up space.
+    #[serde(with = "serde_bytes")]
+    pub pad: Vec<u8>,
+
+    /// Zero-filled bytes used for filling up space.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub pad2: Option<serde_bytes::ByteBuf>,
+
+    /// A human-readable description of what this hash covers.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+
+    /// CBOR byte string describing parameters of the soft binding algorithm.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub alg_params: Option<String>,
+
+    /// A file or http(s) URL to where the bytes that are being hashed lived.
+    ///
+    /// This is useful for cases where the data lives in a different file chunk or side-car
+    /// than the claim.
+    #[deprecated]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub url: Option<UriT>,
+}
+
+/// Details about the soft binding, including the referenced value and scope.
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct SoftBindingBlockMap {
+    /// The scope of the soft binding where it is applicable.
+    pub scope: SoftBindingScopeMap,
+
+    /// In algorithm specific format, the value of the soft binding computed over this block of digital content.
+    pub value: String,
+}
+
+/// Soft binding scope, specifying specifically where in an asset the soft binding is applicable.
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct SoftBindingScopeMap {
+    /// In algorithm specific format, the part of the digital content over which the soft binding value has been computed.
+    #[deprecated]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub extent: Option<String>,
+
+    /// For temporal assets, the timespan in which the soft binding is applicable.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub timespan: Option<SoftBindingTimespanMap>,
+
+    /// Region of interest in regard to the soft binding.
+    pub region: RegionOfInterest,
+}
+
+/// Soft binding timespan for temporal assets.
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct SoftBindingTimespanMap {
+    /// Start of the time range (as milliseconds from media start) over which the soft binding value has been computed.
+    pub start: u32,
```

Comment:
The spec does not specify these to be u32.  Make them usize.  CBOR chooses the optimal storage based on what is actually written.